prboom2/src/MUSIC/portmidiplayer.c

```diff
@@ -118,6 +118,11 @@ static const char *pm_name (void)
 #include <delayimp.h>
 #endif
 
+// converts master volume from linear scaling to non-linear scaling that
+// closely follows vanilla doom behavior but with a smooth transition when
+// approaching maximum volume. cast value to float and divide by 1000.
+const int volume_correction[] = {1000, 1200, 1200, 1200, 1200, 1200, 1200, 1200,
```

Comment:
I still don't get the "cast to float" part. I mean, you are calculating with integers and the result is meant to be an integer, too. If you do all multiplications first and divisions last, the result should be the same, right? 

Apart from this, I am fascinated by your knowledge and expertise of MIDI implementations and would buy about any code change from you. Because, honestly I have no clue about this stuff and what you write just feels right. :grin:

Reply:
Casting to a float here makes the math more decipherable because the related components are connected. Could already have the array be floats with the 1000 divided in though.

Reply:
> Could already have the array be floats with the 1000 divided in though.

Yep, that's be more straightforward.

Reply:
Fixed. Bad habit I picked up from a different project.